tests/_commands/test_train_task.py

```diff
@@ -341,18 +341,47 @@ def test_train_semantic_segmentation__checkpoint(
                     1: "car",
                 },
             },
-            model="dinov2/_vittest14-eomt",
+            model=str(last_ckpt_path),
             model_args={"num_joint_blocks": 1},
             accelerator="auto" if not sys.platform.startswith("darwin") else "cpu",
             devices=1,
             batch_size=2,
             num_workers=0,
             steps=1,
             overwrite=True,
-            checkpoint=last_ckpt_path,
         )
     assert f"Loading model checkpoint from '{last_ckpt_path}'" in caplog.text
 
+    # Part 3: check that the class head can be re-initialized when the number of classes differ.
+    with caplog.at_level(logging.INFO):
+        lightly_train.train_semantic_segmentation(
+            out=out,
+            data={
+                "train": {
+                    "images": train_images,
+                    "masks": train_masks,
+                },
+                "val": {
+                    "images": val_images,
+                    "masks": val_masks,
+                },
+                "classes": {
+                    0: "background",
+                    1: "car",
+                    2: "tree",
+                },
+            },
+            model=str(last_ckpt_path),
+            model_args={"num_joint_blocks": 1},
+            accelerator="auto" if not sys.platform.startswith("darwin") else "cpu",
+            devices=1,
+            batch_size=2,
+            num_workers=0,
+            steps=1,
+            overwrite=True,
+        )
+    assert "Checkpoint provides 3 classes but module expects 4." in caplog.text
```

Comment:
This message is now a bit awkward because the user sets 3 classes in the config and the message complains that there are 3 classes in the model but 4 are expected. --> Should we just remove -1 from the class counts in the message?